ttnn/cpp/ttnn/operations/data_movement/concat/device/concat_device_operation.cpp

```diff
@@ -171,18 +171,26 @@ Tensor concat_impl(std::vector<Tensor> &input_tensors, const std::int64_t dim, c
                         }
                     }
                 }
+                std::variant<int, float> pad_value;
```

Comment:
I think if you used search&replaced and used type alias for `PadValue`/`PadValueType` it'd be a lot cleaner as well.

Reply:
It seems like it's always the same default construction too so you could have default initialization/construction as well to get rid of all this code.

Update: saw later there is use of infinity. Same general recommendation applies I think. Just provide names initializers.

`PadValue pad_value = PadValue::zero();` or `PadValue pad_value = PadValue::neg_infinity();`

Reply:
I minimized my changes now to just tilize_with_val_padding, the changes outside of this to a general PadValue will be outside the scope of this.